### PR TITLE
feat(ffi): API for enabling tracing logs

### PR DESCRIPTION
**Pull request by irvingoujAtDevolution**

### CODE REVIEW

File: ffi/dotnet/Devolutions.IronRdp.AvaloniaExample/MainWindow.axaml.cs [hunk 1]
```diff
@@ -30,6 +30,7 @@ public MainWindow()
 
     private void OnOpened(object? sender, EventArgs e)
     {
+        Log.InitLogWithEnv();
```
Comment (CBenoit) [inline ffi/dotnet/Devolutions.IronRdp.AvaloniaExample/MainWindow.axaml.cs:33]:
```suggestion
        Log.InitLogWithEnv();

```

File: ffi/src/log.rs [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+    #[diplomat::opaque]
+    pub struct Log;
+
+    impl Log {
+        pub fn init_log_with_env() {
+            if LOG_INITED.load(std::sync::atomic::Ordering::Relaxed) {
+                return;
+            }
+            LOG_INITED.store(true, std::sync::atomic::Ordering::Relaxed);
```
Comment (CBenoit) [inline ffi/src/log.rs:18]: suggestion: Use [`Once`](https://doc.rust-lang.org/std/sync/struct.Once.html) instead

File: ffi/src/log.rs [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    use std::{fs::File, path::PathBuf};
+
+    use super::LOG_INITED;
+    use std::fs;
+
+    #[diplomat::opaque]
+    pub struct Log;
+
+    impl Log {
+        pub fn init_log_with_env() {
```
Comment (CBenoit) [inline ffi/src/log.rs:14]: naming:
```suggestion
        pub fn init_with_env() {
```

Will give `Log.InitWithEnv()`

File: ffi/src/log.rs [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+            if LOG_INITED.load(std::sync::atomic::Ordering::Relaxed) {
+                return;
+            }
+            LOG_INITED.store(true, std::sync::atomic::Ordering::Relaxed);
+
+            let filepath: PathBuf = std::env::var("IRONRDP_LOG_FILE")
```
Comment (CBenoit) [inline ffi/src/log.rs:20]: suggestion: Use `IRONRDP_LOG`, like in `ironrdp-client`

File: ffi/src/log.rs [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+                .unwrap_or_else(|_| "/tmp/ironrdp.log".to_owned())
+                .try_into()
+                .expect("Failed to parse log file path");
+
+            fs::create_dir_all(filepath.parent().expect("Path has no parent")).expect("Failed to create log directory");
+            let file = File::create(filepath).unwrap();
+            tracing_subscriber::fmt::SubscriberBuilder::default()
+                .with_env_filter(tracing_subscriber::EnvFilter::from_env("IRONRDP_LOG"))
+                .with_writer(file)
+                .init();
```
Comment (CBenoit) [inline ffi/src/log.rs:30]: suggestion: Reuse the same logic as in `ironrdp-client`

https://github.com/Devolutions/IronRDP/blob/083b5044ad2b961581cf25255aeaf01872edb647/crates/ironrdp-client/src/main.rs#L87-L129

File: ffi/src/log.rs (hunk 2 above)
Comment (irvingoujAtDevolution) [reply ffi/src/log.rs]: updated

File: ffi/src/log.rs (hunk 4 above)
Comment (irvingoujAtDevolution) [reply ffi/src/log.rs]: updated

File: ffi/src/log.rs (hunk 5 above)
Comment (irvingoujAtDevolution) [reply ffi/src/log.rs]: updated

**Review by CBenoit — APPROVED** [review]
LGTM! Thank you!

**Pull request merged**